scripts/cti

```diff
@@ -92,24 +166,67 @@ fi
 
 OUTPUT_TEE=${CTI_OUTPUT_LOG:-$(mktemp)}
 
-echo "**********"
-echo "Copy of this output: $OUTPUT_TEE"
-echo "Dashboard: http://prometheus.${WORKSPACE}.aws.hlw3truzy4ls.com:9091/"
-echo "Logs:"
-echo " * ssh ssh.${WORKSPACE}.aws.hlw3truzy4ls.com"
-echo " * ssh-peer"
-echo " * tail -f /data/libra/libra.log"
-echo "**********"
+if [[ -z "${K8S}" ]]; then
+  if [[ -z "${WORKSPACE}" ]]; then
+    WORKSPACE="cluster-test-ci"
+  fi
+  JUMPHOST=${JUMPHOST:-ssh.${WORKSPACE}.aws.hlw3truzy4ls.com}
 
-if [ "$DEPLOY" == "yes" ]
-then
-    DEPLOY_CMD="--deploy $TAG"
-else
-    DEPLOY_CMD=""
-fi
+  ssh $JUMPHOST echo "ssh ok" >/dev/null || (echo "Failed to ssh to jump host $JUMPHOST. Try renew corp canal cert with cc-certs"; exit 1)
 
+  echo "**********"
+  echo "Copy of this output: $OUTPUT_TEE"
+  echo "Dashboard: http://prometheus.${WORKSPACE}.aws.hlw3truzy4ls.com:9091/"
+  echo "Logs:"
+  echo " * ssh ssh.${WORKSPACE}.aws.hlw3truzy4ls.com"
+  echo " * ssh-peer"
+  echo " * tail -f /data/libra/libra.log"
+  echo "**********"
 
-ssh -t $JUMPHOST ssh -i /libra_rsa ec2-user@ct.priv.${WORKSPACE}.aws.hlw3truzy4ls.com $ENV CTI_MARKER=$MARKER REMOTE_USER=$MARKER ct -c cluster-test-ci --image "853397791086.dkr.ecr.us-west-2.amazonaws.com/libra_cluster_test:$TAG" $DEPLOY_CMD $* | tee $OUTPUT_TEE
+  if [ "$DEPLOY" == "yes" ]
+  then
+      DEPLOY_CMD="--deploy $TAG"
+  else
+      DEPLOY_CMD=""
+  fi
+
+  ssh -t $JUMPHOST ssh -i /libra_rsa ec2-user@ct.priv.${WORKSPACE}.aws.hlw3truzy4ls.com env $ENV CTI_MARKER=$MARKER REMOTE_USER=$MARKER ct -c cluster-test-ci --image "853397791086.dkr.ecr.us-west-2.amazonaws.com/libra_cluster_test:$TAG" $DEPLOY_CMD $* | tee $OUTPUT_TEE
+
+else
+  if ! which kubectl &>/dev/null; then
+    echo "kubectl is not installed. Please install kubectl. On mac, you can use : brew install kubectl"
+    exit 1
+  fi
+  echo "Running cluster-test on Kubernetes"
+  kube_init_context
+  pod_name="cluster-test-$(whoami)-$(date +%s)"
+  specfile=$(mktemp)
+  echo "Pod Spec : ${specfile}"
+  join_args $*
+  join_env_vars $ENV
+  sed -e "s/{pod_name}/${pod_name}/g" \
+      -e "s/{image_tag}/${TAG}/g" \
+      -e "s/{env_variables}/${retval_join_env_vars}/g" \
+      -e "s/{extra_args}/${retval_join_args}/g" \
+      cluster_test_pod_template.yaml > ${specfile}
+  if [[ -z "${WORKSPACE}" ]]; then
+    kube_select_cluster
+    WORKSPACE=${retval_kube_select_cluster}
+  fi
+  echo "Using cluster : ${WORKSPACE}"
+  context=${K8S_CONTEXT_PATTERN/CLUSTERNAME/${WORKSPACE}}
+  kubectl --context=${context} apply -f ${specfile} || (echo "Failed to create cluster-test pod"; exit 1)
+  set +e # kube_wait_pod has a command which can fail
```

Comment:
One alternative would be to rewrite line that might fail with smth like 
```
local phase=$(kubectl --context="${context}" get pod "${pod_name}" -o jsonpath="{.status.phase}" || echo "Failed")
```

`set +e` seem to be very counter intuitive - person looking at this script will have no idea that one function in a file runs with different flags.